src/net/sourceforge/kolmafia/textui/RuntimeLibrary.java

```diff
@@ -4031,6 +4036,37 @@ public static Value adv1(
     return RuntimeLibrary.continueValue();
   }
 
+  private record EightBitZone(DoubleModifier mod, int offset, String color) {}
+
+  private static final Map<Integer, EightBitZone> EIGHT_BIT_ZONES =
+      Map.ofEntries(
+          Map.entry(
+              AdventurePool.FUNGUS_PLAINS, new EightBitZone(DoubleModifier.MEATDROP, 50, "blue")),
+          Map.entry(
+              AdventurePool.HEROS_FIELD, new EightBitZone(DoubleModifier.ITEMDROP, 0, "green")),
+          Map.entry(
+              AdventurePool.VANYAS_CASTLE,
+              new EightBitZone(DoubleModifier.INITIATIVE, 200, "black")),
+          Map.entry(
+              AdventurePool.MEGALO_CITY,
+              new EightBitZone(DoubleModifier.DAMAGE_ABSORPTION, 200, "red")));
+
+  public static Value eight_bit_points(ScriptRuntime controller, final Value locationValue) {
+    var location = (KoLAdventure) locationValue.rawValue();
+
+    var zone = EIGHT_BIT_ZONES.getOrDefault(location.getSnarfblat(), null);
+
+    if (zone == null) {
+      return DataTypes.ZERO_VALUE;
+    }
+
+    var multiplier = zone.color().equals(Preferences.getString("8BitColor")) ? 1 : 0.5;
+    var modValue = KoLCharacter.currentNumericModifier(zone.mod());
+    var points = modValue - zone.offset();
+
+    return new Value(Math.max(0, Math.min(points, 400)) * multiplier);
+  }
```

Comment:
This is confusing. So, let's see.

Bonus points in Vanya's castle maxes out at 600+ initiative and mins out at <= 0 initiative.
(That is an assumption; what happens if you have negative initiative?)
You get either 50 or 100 base points, and up to 150 or 300 bonus points.

With your calculation:

black and Init +0:
multiplier = 1
modvalue = 0
points = 0 - 200 = -200
return max(0, min(-200, 400)) * 1 = 0
--> Base points in a bonus zone is 100, not 0.

black and init +300
multiplier = 1
modValue = 300
points = 100
return max(0, min(100, 400)) * 1 = 100
--> 100 base points 150 bonus points = 250 points

black and Init +600
multiplier = 1
modvalue = 600
points = 600 - 200 = 400
return max(0, min(400, 400)) * 1 = 400
--> this is correct: 100 base points and 300 bonus.

Reply:
Perhaps something like this would be simpler?
```
private record EightBitZone(DoubleModifier mod, int points, String color) {}

private static final Map<Integer, EightBitZone> EIGHT_BIT_ZONES =
       Map.ofEntries(
           Map.entry(
               AdventurePool.FUNGUS_PLAINS, new EightBitZone(DoubleModifier.MEATDROP, 2, "red")),
           Map.entry(
               AdventurePool.HEROS_FIELD, new EightBitZone(DoubleModifier.ITEMDROP, 3, "green")),
           Map.entry(
               AdventurePool.VANYAS_CASTLE,
               new EightBitZone(DoubleModifier.INITIATIVE, 1, "black")),
           Map.entry(
               AdventurePool.MEGALO_CITY,
               new EightBitZone(DoubleModifier.DAMAGE_ABSORPTION, 1, "blue")));

var multiplier = zone.color().equals(Preferences.getString("8BitColor")) ? 1 : 0.5;
var modValue = KoLCharacter.currentNumericModifier(zone.mod());
var bonus = (modValue - 300) * zone.points();

return new Value(multiplier * (100 + Math.min(300, Math.max(0, bonus))));
```
For mods over 300: bonus is 2 points per Meat Drop, 3 per Item Drop, or 1 for DA or Init.